File: 2145-grid-game/grid-game.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var gridGame = function (grid) {
    let topSum = 0
    let bottomSum = 0
    let robot2 = Infinity
    for (let i = 0; i < grid[0].length; i++) {
        topSum += grid[0][i]
    }
    for (let i = 0; i < grid[0].length; i++) {
        topSum -= grid[0][i]
        robot2 = Math.min(robot2, Math.max(topSum, bottomSum))
        bottomSum += grid[1][i]
    }
    return robot2
};

File: 2145-grid-game/grid-game.py
class Solution:
    def gridGame(self, grid: List[List[int]]) -> int:
        topSum = 0
        bottomSum = 0
        robo2 = float('inf')
        for i in range(len(grid[0])):
            topSum += grid[0][i]
        for i in range(len(grid[0])):
            topSum -= grid[0][i]
            
            robo2 = min(robo2,max(topSum,bottomSum))
            bottomSum += grid[1][i]
        return robo2